analyses/independent-samples/independent_rna_samples.R

```diff
@@ -0,0 +1,104 @@
+# independent_rna_samples.R
+
+#' Generate a vector of unique rna samples
+#' 
+#' The samples from this function will be unique with respect to participants
+#' i.e. only no two samples will come from the same participant. The input list 
+#' should be pre-filtered by `composition` and `sample_type`.  
+#' 
+#' 
+#' @param independent_dna_sample_df A data frame of samples, with columns 
+#' corresponding to those in `independent-specimens.wgswxs.primary.tsv` 
+#' or `independent-specimens.wgswxs.primary-plus.tsv` depending on what 
+#' set of samples you need to include
+#' @param histology_df A data frame of samples, with columns corresponding 
+#' to those `pbta-histologies.tsv`
+#' @param match_type Designates which type matching needs to be done. Options
+#' are "independent_dna" to include only rna samples that match the 
+#' independent-specimens.wgswxs sample set, 
+#' "independent_dna_plus_only_rna" to include samples that macth the dna sample 
+#' set plus include samples where only rna samples exists
+#' @param tumor_description_rna_only Tumor descriptors to select samples where
+#' only RNA samples are available and will have no matching id in independent_dna_sample_df
+#' Opetions are "primary" to select only primary/initial tumors As of v18, primary tumors are defined as those designated "Initial CNS Tumor" .
+#' "primary_plus" if you would like to select other non-initial tumor RNA-Seq sample if no 
+#' initial tumor RNA-Seq sample exists
+#' or "Diagnosis" in the `tumor_descriptor` field.
+#' @param seed An optional random number seed. 
+#' 
+#' @return a data frame of Participant and Specimen IDs, each present only once.
+independent_rna_samples <- function(independent_dna_sample_df, 
+                                histology_df,
+                                match_type = c("independent_dna", "independent_dna_plus_only_rna"),
+                                tumor_description_rna_only = c("primary","primary_plus"),
+                                seed){
+  match_type <- match.arg(match_type)
+  tumor_description_rna_only <- match.arg(tumor_description_rna_only)
+  if(!missing(seed)){set.seed(seed)}
+  primary_descs <- c("Initial CNS Tumor", "Diagnosis")
+  
+  
+  if(match_type == "independent_dna" | match_type == "independent_dna_plus_only_rna"){
```

Comment:
This is all cases, correct? If so, we don't need the `if` statement, which confused me on initial reading.

Reply:
Yes, I removed the if statement here as well..